Factory/HttpFoundationFactory.php

```diff
@@ -28,6 +28,16 @@
  */
 class HttpFoundationFactory implements HttpFoundationFactoryInterface
 {
+    protected $requestClass;
+
+    public function __construct($requestClass = null)
+    {
+        $this->requestClass = $requestClass;
+        if ($this->requestClass === null) {
```

Comment:
What about checking that the class given is an actual subclass of the `Request` class from the HttpFoundation component?